util/compression_test.cc

```diff
@@ -1768,122 +1768,219 @@ TEST_F(DBAutoSkip, AutoSkipCompressionManager) {
+    options_.rate_limiter.reset(NewGenericRateLimiter(
+        1000000000, 1000 /* refill_period_us */, 10 /* fairness */,
+        RateLimiter::Mode::kWritesOnly));
+    std::shared_ptr<IOGoal> io_goal =
+        std::make_shared<IOGoal>(kIOUpperBound, kIOLowerBound);
+    std::shared_ptr<IOGoal> cpu_budget =
+        std::make_shared<IOGoal>(kCPUUpperBound, kCPULowerBound);
+    options_.compression_manager = CreateAutoTuneCompressionManager(
+        nullptr, io_goal, cpu_budget, options_);
+    DestroyAndReopen(options_);
   }
-  const int kValueSize = 20000;
-  int next_key = 0;
-  Random rnd(231);
-  auto value = rnd.RandomBinaryString(kValueSize);
-  int window_size = 10;
-  auto WindowWrite = [&]() {
-    for (auto i = 0; i < window_size; ++i) {
-      auto status = Put(Key(next_key), value);
+  void BlockWrite(int num) {
+    constexpr int kValueSize = 20000;
+    auto value = rnd.RandomBinaryString(kValueSize);
+    for (auto i = 0; i < num; ++i) {
+      auto status = Put(Key(next_key_), value);
       EXPECT_OK(status);
-      next_key++;
+      next_key_++;
     }
   };
-  // This denotes the first window
-  // Mocked to have specific cpu utilization and io cost
-  WindowWrite();
-  // check the predictor is predicting the correct cpu and io cost
-  WindowWrite();
-  ASSERT_OK(Flush());
+};
+TEST_F(DBAutoTuneCompressionTest, AutoTuneCompression) {
+#ifdef _WIN32
+  // Skip this test on Windows platforms as the method of measuring CPU usage is
+  // not implemented yet on the windows platform
+  return;
+#endif
+  // make sure that threre are more than two compressors before running the
+  // test case.
+  auto supported_compressions = GetSupportedCompressions();
+  // Check if KLZ4Compression, KLZ4HCCompression and kZSTDCompression are
+  // supported before running the test case as they must be supported for us to
+  // have at least two compressors
+  if (std::find(supported_compressions.begin(), supported_compressions.end(),
+                kLZ4Compression) != supported_compressions.end() ||
+      std::find(supported_compressions.begin(), supported_compressions.end(),
+                kLZ4HCCompression) != supported_compressions.end() ||
+      std::find(supported_compressions.begin(), supported_compressions.end(),
+                kZSTD) != supported_compressions.end()) {
+    // Skipping since none of the compression is supported
+
+    auto default_type =
+        supported_compressions[supported_compressions.size() - 1];
+    options_.compression = default_type;
+    DestroyAndReopen(options_);
+    // Setting up the condition to mock CPU IO Usage and CPU IO Cost prediction
+    size_t cur_selection = 0;
+    size_t default_cpu_prediction = 500;
+    size_t default_io_prediction = 1000;
+    size_t expected_sel_cpu_prediction = 700;
+    size_t expected_sel_io_prediction = 700;
+    size_t expected_selection = 1;
+    auto cpu_usage = 0.5;
+    auto io_usage = 0.5;
```

Comment:
Variables about cpu, io need to be clear about their units and conssitent. In public API, its bytes and ratio of core. In db bench, it's ratio of rate limiter throughput and ratio and core. Here this appear to be something else. Please consolidate.